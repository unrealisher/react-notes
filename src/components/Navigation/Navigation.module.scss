.navigation {
  background-color: white;
}

.menu {
  display: flex;
  flex-direction: column;
  padding: 0 16px;
  padding-bottom: 15px;
}

.item {
  width: 100%;
  text-align: center;
}

.button {
  cursor: pointer;
  padding: 10px 0;
  width: 100%;
  border: none;
  background-color: white;
  font-size: 16px;
  line-height: 18px;

  &_add {
    border-radius: 4px;
    background-color: #fc0;
    margin-bottom: 15px;
  }
}

@media (min-width: 970px) {
  .menu {
    flex-direction: row;
    padding: 0;
  }

  .item_archive {
    margin-right: 16px;
  }

  .button {
    padding: 10px 16px;
    margin: 0;
  }
}
