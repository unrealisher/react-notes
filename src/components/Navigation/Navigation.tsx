import React from "react";

import Search from "./../Search/Search";

import styles from "./Navigation.module.scss";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { setActiveNotes } from "../../store/actions/setActiveNotes";
import { setArchiveNotes } from "../../store/actions/setArchiveNotes";

interface IDispatchToProps {
  onActiveClick: Function;
  onArchiveClick: Function;
}

interface IProps extends IDispatchToProps {
  setPopup: Function;
}

const Navigation = (props: IProps): JSX.Element => {
  const { onActiveClick, onArchiveClick, setPopup } = props;
  return (
    <nav className={styles.navigation}>
      <h2 hidden>Главное меню</h2>
      <ul className={styles.menu}>
        <li className={styles.item}>
          <button className={styles.button} onClick={() => onActiveClick()}>
            Активные
          </button>
        </li>
        <li className={styles.item + " " + styles.item_archive}>
          <button className={styles.button} onClick={() => onArchiveClick()}>
            Архив
          </button>
        </li>
        <li className={styles.item}>
          <button
            className={styles.button + " " + styles.button_add}
            onClick={() => setPopup(true)}
          >
            Добавить
          </button>
        </li>
        <li className={styles.item}>
          <Search />
        </li>
      </ul>
    </nav>
  );
};

const mapDispatchToProps = (dispatch: Dispatch): IDispatchToProps => {
  return {
    onActiveClick: (): void => {
      dispatch(setActiveNotes());
    },
    onArchiveClick: (): void => {
      dispatch(setArchiveNotes());
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(Navigation);
