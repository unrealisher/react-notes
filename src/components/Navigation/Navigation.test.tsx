import React from "react";
import { shallow } from "enzyme";
import { Navigation } from "./Navigation";

describe("Navigation", () => {
  const onActiveClick = (): void => {};
  const onArchiveClick = (): void => {};
  const onFetchArchive = (): void => {};
  const setPopup = (): void => {};

  it("correct render", () => {
    const component = shallow(
      <Navigation
        onActiveClick={onActiveClick}
        onArchiveClick={onArchiveClick}
        onFetchArchive={onFetchArchive}
        setPopup={setPopup}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("onActiveClick calls when button is clicked", () => {
    const onActiveClick = jest.fn();
    const component = shallow(
      <Navigation
        onActiveClick={onActiveClick}
        onArchiveClick={onArchiveClick}
        onFetchArchive={onFetchArchive}
        setPopup={setPopup}
      />
    );
    const button = component.find(".button").first();
    button.simulate("click");
    expect(onActiveClick).toBeCalled();
  });

  it("setPopup calls when button is clicked", () => {
    const setPopup = jest.fn();
    const component = shallow(
      <Navigation
        onActiveClick={onActiveClick}
        onArchiveClick={onArchiveClick}
        onFetchArchive={onFetchArchive}
        setPopup={setPopup}
      />
    );
    const button = component.find(".button_add");
    button.simulate("click");
    expect(setPopup).toBeCalledWith(true);
  });

  it("onArchiveClick calls when button is clicked", () => {
    const onArchiveClick = jest.fn();
    const component = shallow(
      <Navigation
        onActiveClick={onActiveClick}
        onArchiveClick={onArchiveClick}
        onFetchArchive={onFetchArchive}
        setPopup={setPopup}
      />
    );
    const button = component.find(".item_archive button");
    button.simulate("click");
    expect(onArchiveClick).toBeCalled();
  });

  it("onFetchArchive calls when button is clicked", () => {
    const onFetchArchive = jest.fn();
    const component = shallow(
      <Navigation
        onActiveClick={onActiveClick}
        onArchiveClick={onArchiveClick}
        onFetchArchive={onFetchArchive}
        setPopup={setPopup}
      />
    );
    const button = component.find(".item_archive button");
    button.simulate("click");
    expect(onFetchArchive).toBeCalled();
  });
});
