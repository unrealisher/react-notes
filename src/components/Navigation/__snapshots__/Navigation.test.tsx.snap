// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    onActiveClick={[Function]}
    onArchiveClick={[Function]}
    onFetchArchive={[Function]}
    setPopup={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          hidden={true}
        >
          Главное меню
        </h2>,
        <ul
          className="menu"
        >
          <li
            className="item"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              Активные
            </button>
          </li>
          <li
            className="item item_archive"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              Архив
            </button>
          </li>
          <li
            className="item"
          >
            <button
              className="button button_add"
              onClick={[Function]}
            >
              Добавить
            </button>
          </li>
          <li
            className="item"
          >
            <Connect(Search) />
          </li>
        </ul>,
      ],
      "className": "navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Главное меню",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Главное меню",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="item"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                Активные
              </button>
            </li>,
            <li
              className="item item_archive"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                Архив
              </button>
            </li>,
            <li
              className="item"
            >
              <button
                className="button button_add"
                onClick={[Function]}
              >
                Добавить
              </button>
            </li>,
            <li
              className="item"
            >
              <Connect(Search) />
            </li>,
          ],
          "className": "menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button"
                onClick={[Function]}
              >
                Активные
              </button>,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Активные",
                "className": "button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Активные",
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button"
                onClick={[Function]}
              >
                Архив
              </button>,
              "className": "item item_archive",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Архив",
                "className": "button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Архив",
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button button_add"
                onClick={[Function]}
              >
                Добавить
              </button>,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Добавить",
                "className": "button button_add",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Добавить",
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Search) />,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            hidden={true}
          >
            Главное меню
          </h2>,
          <ul
            className="menu"
          >
            <li
              className="item"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                Активные
              </button>
            </li>
            <li
              className="item item_archive"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                Архив
              </button>
            </li>
            <li
              className="item"
            >
              <button
                className="button button_add"
                onClick={[Function]}
              >
                Добавить
              </button>
            </li>
            <li
              className="item"
            >
              <Connect(Search) />
            </li>
          </ul>,
        ],
        "className": "navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Главное меню",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Главное меню",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="item"
              >
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Активные
                </button>
              </li>,
              <li
                className="item item_archive"
              >
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Архив
                </button>
              </li>,
              <li
                className="item"
              >
                <button
                  className="button button_add"
                  onClick={[Function]}
                >
                  Добавить
                </button>
              </li>,
              <li
                className="item"
              >
                <Connect(Search) />
              </li>,
            ],
            "className": "menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="button"
                  onClick={[Function]}
                >
                  Активные
                </button>,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Активные",
                  "className": "button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Активные",
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="button"
                  onClick={[Function]}
                >
                  Архив
                </button>,
                "className": "item item_archive",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Архив",
                  "className": "button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Архив",
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="button button_add"
                  onClick={[Function]}
                >
                  Добавить
                </button>,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Добавить",
                  "className": "button button_add",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Добавить",
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Search) />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
